### PR TITLE
Created new sub category in resources

### LINKED ISSUE #2107: [Self Paced Courses],  New Sub category in Resources
### Category Name

Practice Website

### Subcategory Name

Freecodecamp

### Additional Context

There are lots of useful websites that let you code with in-browser IDEs such as FreecodeCamp and many more!! which can help beginners with industrial practices.

And can you suggest any good category name for this?
Thank You

### Checklist

- [X] I have checked the existing [issues](https://github.com/rupali-codes/LinksHub/issues?q=is%3Aissue+)
- [X] I have read the [Contributing Guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md)
- [X] I have checked that this category does not exist
- [X] I have checked that this subcategory does not exist
- [X] I am willing to work on this issue (optional)
- [X] I am participating in Hacktoberfest 2023

### PR DESCRIPTION
**Pull request by Amanmawar17**
## Fixes Issue  

closes #2107

## Changes proposed
Added a new sub-category in the resources tab which was self-paced courses.

I made a new JSON file in the resources folder named self-paced courses.
I committed changes data.ts and index.ts to import new self-paced-course.json for this to be shown in the category.
## Screenshots
Screenshot 2023-10-27 125930

## Note to reviewers
Suggest any required changes!!
Thank You.

### CODE REVIEW

**Comment by Anmol-Baranwal** [comment]
@Amanmawar17 
Following the PR template is a must. It includes special words like 'fixes' or 'resolves' that connect your issue to the PR, as explained in the official docs.

If you don't link the issue correctly, even after the PR is merged, the issue might stay open, causing huge problems.

Including it in title won't work. 

You can see here, if the issue is linked.
image

I've done it for you.

Just a heads-up to help you link issue properly next time. Have a great day!

**Review by Anmol-Baranwal — CHANGES_REQUESTED** [review]
@Amanmawar17 
Please add the description of the subcategory.

It will be in `TopBar/CategoryDescriptions.ts`

**Comment by Amanmawar17** [comment]
> @Amanmawar17 Please add the description of the subcategory.
> 
> It will be in `TopBar/CategoryDescriptions.ts`

I resolved this issue but can you tell me how I made a PR for new changes?

File: database/resources/self-paced-courses.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  {
+    "name": "FreeCodeCamp",
+    "description": "We can learn various self paced beginner friendly courses in their own website code editor.",
+    "url": "https://www.freecodecamp.org/learn",
+    "category": "resources",
+    "subcategory": "self-paced-courses"
+  },
+  {
+    "name": "Exercism",
+    "description": "This website provides 67 programming languages support to develop fluency in their very own way.",
```
Comment (CBID2) [inline database/resources/self-paced-courses.json:11]:
```suggestion
    "description": "This website provides 67 programming languages to help aspiring developers to develop fluency in their own way.",
```

File: database/resources/self-paced-courses.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    "url": "https://exercism.org/tracks",
+    "category": "resources",
+    "subcategory": "self-paced-courses"
+  },
+  {
+    "name": "CodeCademy",
+    "description": "Learning and mastery the both languages and various tracks that available on this website.",
```
Comment (CBID2) [inline database/resources/self-paced-courses.json:18]:
```suggestion
    "description": "Learning and master both languages in various tracks on this website.",
```

File: database/resources/self-paced-courses.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
+[
+  {
+    "name": "FreeCodeCamp",
+    "description": "We can learn various self paced beginner friendly courses in their own website code editor.",
```
Comment (CBID2) [inline database/resources/self-paced-courses.json:4]:
```suggestion
    "description": "Learn how to code and in various self paced beginner friendly courses on Codecademy. You can also put your skills into practice with their own website code editor.",
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @Amanmawar17. The descriptions in your PR needs more improvement, so I provided suggestions to help you. Other than that, you're off to a great start! 😊

**Review by Anmol-Baranwal — COMMENTED** [review]
It's advisable to use "e-book" or "ebook" to properly convey the meaning of an electronic book.

What do you think?  `e book`, `e-book`, `ebook`

image

image

I am confused about the extra change. The description is definitely fixed, but which one is more suitable?

**Comment by Amanmawar17** [comment]
if we choose E-book then the description would not be there, and also E-Book is also not there,  showing this 
Screenshot 2023-10-29 112542

so what revert back this E-book change and then open another issue for it?

**Review by Anmol-Baranwal — DISMISSED** [review]
🚀 Looks good to me. 

@Amanmawar17 
It's fine, we will change it if it feels odd later on.

**Comment by Anmol-Baranwal** [comment]
@CBID2 
You're up for review!

**Review by CBID2 — DISMISSED** [review]
Looks great

**Comment by CBID2** [comment]
Found some conflicts that need merging @Amanmawar17

**Comment by Anmol-Baranwal** [comment]
@Amanmawar17 
You can see the changes, and push it accordingly if you want, it will fix merge conflicts in case you have trouble using command line for the same.

**Review by rupali-codes — APPROVED** [review]
lgtm

**Comment by rupali-codes** [comment]
HI @Amanmawar17 please resolve the merge conflicts

**Review by rupali-codes — APPROVED** [review]
Lgtm

**Comment by rupali-codes** [comment]
@Amanmawar17  please resolve merge conflicts

**Comment by Anmol-Baranwal** [comment]
I'm closing this issue due to inactivity. Please feel free to reopen it if you'd like to contribute or have further discussions.

**Pull request closed without merging**